StudentAppMvc/StudentAppMvc/Controllers/SubjectController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using StudentAppMvc.Exceptions;
+using StudentAppMvc.Filter;
+using StudentAppMvc.Models;
+using StudentAppMvc.Models.ViewModels;
+using StudentAppMvc.Services;
+
+namespace StudentAppMvc.Controllers
+{
+    //[AuthenticationFilter]
+    [LogginFilter]
+    public class SubjectController : Controller
+    {
+        public static List<Subject> _subjectList;
+        private ISubjectService _subjectService;
+
+        
+        public SubjectController(ISubjectService subjectService)
+        {
+            _subjectService = subjectService;
+        }
+
+        // GET: StudentController
+        public ActionResult Index()
+        {
+            SubjectViewModel subjectViewModel = new SubjectViewModel(_subjectService.ListSubject(), "my testin");
+            return View(subjectViewModel);
+        }
+
+
+        // GET: StudentController/Create
+        public ActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: StudentController/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Create(Subject subject)
+        {
+            try
+            {
+                _subjectService.Create(subject);
```

Comment:
Should use the following validation lines before calling service

if (!ModelState.IsValid)
    {
        return Page();
    }